### DISCUSSION TITLE
Is an `ExitBlock` a `BasicBlock`?

### DISCUSSION THREAD

**Discussion by cqc-alec** [Quantinuum/hugr#155] [Specification]
In the current implementation, there are two kinds of `BasicBlock`: `Block` and `Exit`. In the spec, `BasicBlock` and `ExitBlock` are distinguished:

> All children except the second are BasicBlock-nodes, the first is the entry block. The second child is an ExitBlock node

Which is the more natural approach?

**Comment by ss2165** [comment]
I think this is a misaligned usage of "BasicBlock" in the spec and  the implementation.
Essentially BasicBlock(Spec) = BasicBlock::Block(Implementation). 

I think the implementation one is nicer. But "Block::Block" isn't very nice. 
How about converging on  "Basic::{Block, Exit}" ? 
I.e. `Block` and `Exit` are both `Basic`, `CFG` nodes have only `Basic` children. The second child must be `Exit`, all others are `Block`.

**Reply by cqc-alec** [reply]
Sounds good to me; I guess you mean "`CFG` nodes can have only `Basic` children".

**Reply by ss2165** [reply]
yes, edited

**Reply by cqc-alec** [reply]
Was about to convert this to an Issue, but then thought, feels a bit of a cop-out; "Basic" is an adjective and it would be better to have a noun for the node type. But it's hard to think of one that covers both "basic block" and "exit node".

We could stick with what we have in the implementation but rename the "Block" subtype to "Process": so "BasicBlock::{Process, Exit}"?

**Reply by ss2165** [reply]
What about `BasicBlock::DFB` for "Dataflow Block"

**Reply by cqc-alec** [reply]
That works.

**Comment by cqc-alec** [comment]
#171 #172
